### PR TITLE
RichText: show boundary only with editable element focus

### LINKED ISSUE #15463: Rich Text: Non-selected inline boundaries wrongly highlighted on blur
Related: #14120 (cc @ellatrix )

**Describe the bug**

Similar to what was demonstrated in #14120, it can occur that all inline boundaries of a kind appear as highlighted when deselecting a paragraph while the caret is within an inline boundary.

**To reproduce**

1. Navigate to Posts > Add New
2. Add a paragraph with bolded text
3. Add another paragraph with bolded text
4. Click within one or the other of the bolded fragments from Step 2 or Step 3
5. Click in the gutters to the left or right of the block list to deselect the current paragraph
6. Note that both bolded texts appear as highlighted

**Expected behavior**

Neither bolded text should appear as highlighted.

**Screenshots**

highlight

**Desktop (please complete the following information):**

 - OS: macOS Mojave 10.14.4 (18E226)
 - Browser: Version 74.0.3729.131 (Official Build) (64-bit)
- Gutenberg version: `wordpress-develop@96649a5` (effectively WordPress 5.2-RC.2)

**Additional context**

It seems to be important that the highlight had at one point previously been highlighted, in order for it to appear as highlighted when deselecting the block.

### PR DESCRIPTION
**Pull request by ellatrix**
## Description

Fixes #15463. The focus check should be more specific to editable elements. If a parent element has focus, e.g. the block or body element, the boundary should not be visible.

Of course the boundary attribute should be removed on blur, but we're not ready yet to remove "lingering" selection state on blur (e.g. formatting buttons would not work).

## How has this been tested?
See #15463.

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/block-editor/src/components/rich-text/index.js [hunk 1]
```diff
@@ -493,9 +493,10 @@ export class RichText extends Component {
 			const newColor = computedStyle.color
 				.replace( ')', ', 0.2)' )
 				.replace( 'rgb', 'rgba' );
+			const path = `[contenteditable]:focus ${ boundarySelector }`;
```
Comment (aduth) [inline packages/block-editor/src/components/rich-text/index.js]: Is there a reason it not be more specific to a `contenteditable` rendered in `.editor-rich-text` ? While certainly an edge case, would I be wrong in thinking there's still the potential here for some false positives with e.g. `<div contentEditable><RichText /></div>` ?

File: packages/block-editor/src/components/rich-text/index.js [hunk 2]
```diff
@@ -493,9 +493,10 @@ export class RichText extends Component {
...
+			const rules = `{background-color: ${ newColor }}`;
 
-			globalStyle.innerHTML =
-				`*:focus ${ boundarySelector }{background-color: ${ newColor }}`;
+			globalStyle.innerHTML = path + rules;
```
Comment (aduth) [inline packages/block-editor/src/components/rich-text/index.js]: The composing of this string is a bit hard to follow for me, likely a consequence of the placement of `{` (not helped by template strings also having the `{`). Is the purpose to have shorter lines? I'd be more okay with the longer line, personally. There's also an opportunity for an early return for the negated condition, to de-indent by one.

```js
if ( ! element ) {
	return;
}
```

**Review by aduth — DISMISSED** [review]
In my testing, this seems to work a little too well, i.e. I can't get the inline boundary to show again when clicking back inside the element.

highlight-2

Generally I'm not clear why either the style can't be assigned directly (related to #13838?) or the style selector be more specific to guarantee that we're targeting the intended element (is `data-rich-text-format-boundary` removed when changing a selection to guarantee we can't have two highlighted in the same field?).

**Comment by aduth** [comment]
Oddly, in the prior comment, the caret is within the bolded text, but it is not marked up as `data-rich-text-format-boundary`:

image

File: packages/block-editor/src/components/rich-text/index.js (hunk 1 above)
Comment (ellatrix) [reply packages/block-editor/src/components/rich-text/index.js]: Yeah, we should use the class.

**Comment by ellatrix** [comment]
@aduth Yes, something else seems to be going wrong there. :/ I'll need to have a look. Also, when you use the arrow keys inside formatting, the boundary disappears.

**Review by aduth — APPROVED** [review]
Wonder if it's the sort of thing we could have end-to-end tests for? I'm okay as a follow-up task.

Otherwise, it's working well for me 👍
